oslabs-beta/vista
# Nestorcayananjr/refactoring react flow

# Issue Type


# Issue Description & Business Use Case

Refactored React Flow portion of application including...

- Breaking down DisplayData component into two components: DisplaySchemaContainer(container) and DisplaySchema(presentational).
- Abstracted business logic from DisplayData into a new file: createNewNodesAndEdges.ts
- Refactored how new nodes and edges were being added. Now is utilizing React Flow's built in useNodes/EdgesState hook.
- Linting/cleaning of various parts of code.

# Ticket Item

N/A

# Steps to Reproduce Bug / Validate Feature / Confirm Tech Debt Fix

1. Pull latest changes
2. Observe file structure, verifying three new files were created: DisplaySchema, DisplaySchemaContainer, and createNewNodesAndEdges
3. Run Dev Server
4. Verify application is running as normal

# Previous behavior

No visual changes in behavior. State management of nodes should improve and testing should be more effected with modularized components and abstracted logic.

# Expected behavior

Application should run as expected
